Feat/#56 고민 상세 페이지 - 해결 후 API 연결

wkdyujin opened the pull request:
<!-- 🔥 Assignee, Label, Reviewer 설정!!!🔥 -->

## 📒 이슈

- Resolved: #56

## 🔥 작업 내용

- 고민글 삭제, 최종 결정 후 나의 기록 API 연결했습니다.

## ☑️ PR Point

<!-- 피드백을 받고 싶은 부분, 공유하고 싶은 부분, 작업 과정, 이유를 적어주세요. -->

 - 고민글 삭제에서 warning 다이얼로그 사용하려고 했지만 다음의 이유로 UI 따로 생성했습니다. 1)데이터바인딩 표현식 사용하면 코드상에서 text를 변경 불가 / 2) yes, no 위치 반대 / 3) java test.html deprecated
 - 고민글 리뷰 등록에서 edittext 포커싱 시 키보드가 글자를 가리는 이슈가 있습니다,, 스크롤뷰에 padding 추가, smoothScrollTo로 edittext 위치 아래까지 스크롤되게 해봐도 반영이 안 됩니다😂
 - activity_detail에 포함된 layout_detail_answer 요소들이 중앙 정렬 되어있어서 좌측 정렬로 수정했씁니다.

- 고민글 삭제는 통신 응답으로 500 와서 확인은 못해봤습니다. (스웨거도 500)

## 📱실행결과

<!-- gif or mp4. gif는 [https://ezgif.com/](https://ezgif.com/) 활용! 용량제한 10MB넘어가면 카톡으로.. -->
고민글 삭제
https://github.com/TeamHARA/KAERA_Android/assets/69359774/1f077b73-b72a-417c-9733-cf8b697f69f1


고민 리뷰 수정
https://github.com/TeamHARA/KAERA_Android/assets/69359774/d01f35d9-9203-4e95-9da3-eecf72cedc97

IslandofDream committed:
[Feat] 다이얼로그 팝업 애니메이션 추가

wkdyujin committed:
[Feat] 고민글 삭제 다이얼로그 UI 추가

wkdyujin committed:
[Feat] 고민글 삭제 Api 연결
- 500으로 응답 오는 것 확인

wkdyujin committed:
[Feat] 리뷰 저장 UI 추가 및 질문과 응답 레이아웃 수정

wkdyujin committed:
[Feat] 나의 기록 등록 및 수정하기 API 연결

wkdyujin committed:
[Feat] 고민 리뷰 작성 완료 다이얼로그 추가

IslandofDream commented on app/src/main/java/com/hara/kaera/presentation/detail/DetailAfterActivity.kt:
```diff
@@ -0,0 +1,157 @@
+    }
+
+    private fun collectFlows() {
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.detailStateFlow.collect {
+                    render(it)
+                }
+            }
+        }
+
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.deleteWorryFlow.collect {
+                    renderDelete(it)
+                }
+            }
+        }
+
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.reviewWorryFlow.collect {
+                    renderUpdateReviewDate(it)
+                }
+            }
+        }
```
```suggestion
        lifecycleScope.launch {
            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
                launch {
                    viewModel.detailStateFlow.collect {
                        render(it)
                    }   
                }
                launch {
                    viewModel.deleteWorryFlow.collect {
                        renderDelete(it)
                    }
                }
                launch {
                    viewModel.reviewWorryFlow.collect {
                        renderUpdateReviewDate(it)
                    }
                }
            }
        }
```
이렇게 하면 줄어들듯?

wkdyujin replied:
repeatOnLifecycle 한 번만 열고 flow를 각각 다른 코루틴에서 수집하면 되는구나
반영할게요 감사합니다☺️

IslandofDream commented on app/src/main/java/com/hara/kaera/presentation/storage/viewmodel/StorageViewModel.kt:
```diff
@@ -37,13 +38,16 @@ class StorageViewModel @Inject constructor(
+                        is ApiResult.Success -> {
+                            _worryStateFlow.value = UiState.Success(collect.data)
+                        }
+                        is ApiResult.Error -> {
+                            _worryStateFlow.value = UiState.Error(collect.error.toString())
```
```suggestion
                        is ApiResult.Error -> {
                            _worryStateFlow.value = UiState.Error(errorToMessage(collect.error))
```

wkdyujin replied:
잊어버리고 사용을 안 했네.. !!! 당장 넣을게요

IslandofDream commented on app/src/main/java/com/hara/kaera/presentation/detail/DetailAfterViewModel.kt:
```diff
@@ -0,0 +1,105 @@
+
+@HiltViewModel
+class DetailAfterViewModel @Inject constructor(
+    private val useCase: GetWorryDetailUseCase,
```
useCase 이름 worryUseCase이정도로 하면좋을듯?

IslandofDream commented on app/src/main/java/com/hara/kaera/presentation/detail/DetailAfterViewModel.kt:
```diff
@@ -0,0 +1,105 @@
+                        }
+
+                        is ApiResult.Error -> {
+                            _reviewWorryFlow.value = UiState.Error(collect.error.toString())
```
```suggestion
                            _reviewWorryFlow.value = UiState.Error(errorToMessage(collect.error))
```

IslandofDream reviewed (commented):
Put이나 Delete 는 사용자 추가되고 한번 더 보면 되겠다!👍

wkdyujin committed:
Update app/src/main/java/com/hara/kaera/presentation/detail/DetailAfterActivity.kt

코드리뷰 반영 리팩토링

Co-authored-by: JunwooKim <rhffldkt0517@naver.com>

wkdyujin committed:
Update app/src/main/java/com/hara/kaera/presentation/storage/viewmodel/StorageViewModel.kt

코드리뷰 반영

Co-authored-by: JunwooKim <rhffldkt0517@naver.com>

wkdyujin committed:
Update app/src/main/java/com/hara/kaera/presentation/detail/DetailAfterViewModel.kt

코드리뷰 반영

Co-authored-by: JunwooKim <rhffldkt0517@naver.com>

wkdyujin committed:
[Refactor] 네이밍 변경

wkdyujin committed:
[Refactor] errorToMessage 반영

IslandofDream committed:
[Refactor] 애니메이션 페이드 인 아웃으로 변경 및 어색한 다이얼로그 애니메이션 수정
- <item name="android:windowIsFloating">false</item>
얘 때문에 다이얼로그가 나타날때 어색한 애니메이션이 나왔음

skylartosf reviewed (approved):
✨서버통신 최고

wkdyujin committed:
[Fix] 고민 이후 나의 기록 edittext가 키보드에 가려지는 문제 해결

wkdyujin committed:
Merge branch 'develop' of https://github.com/TeamHARA/KAERA_Android into feat/#56

 Conflicts:
	app/src/main/java/com/hara/kaera/data/mapper/Mapper.kt

wkdyujin committed:
[Refactor] Empty UiState 추가로 인한 리팩토링

wkdyujin committed:
[Refactor] 코드 정리

wkdyujin committed:
Merge branch 'develop' of https://github.com/TeamHARA/KAERA_Android into feat/#56

 Conflicts:
	app/src/main/java/com/hara/kaera/data/mapper/Mapper.kt
